### PR TITLE
Add NetworkMonitor

### PR DESCRIPTION
**Pull request by maraoz**

### CODE REVIEW

File: lib/eventbus.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ function EventBus() {
 util.inherits(EventBus, EventEmitter);
 
 EventBus.prototype.process = function(e) {
-  $.checkArgument(_.isObject(e));
+  $.checkArgument(_.isObject(e), 'Expected object, found '+ e);
```
Comment (eordano) [inline lib/eventbus.js]: `e + '(' + typeof(e) + ')'`?

File: lib/networkmonitor.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+var Networks = bitcore.Networks;
+var p2p = require('bitcore-p2p');
+var Promise = require('bluebird');
+var $ = bitcore.util.preconditions;
+var _ = bitcore.deps._;
+var EventEmitter = require('events').EventEmitter;
+var util = require('util');
+var EventBus = require('./eventbus');
+var Peer = p2p.Peer;
+var Messages = p2p.Messages;
```
Comment (yemel) [inline lib/networkmonitor.js]: Could you sort this list? A good practice is to separate it in 3 groups:

1 - Third party libraries
2 - Our external libraries (bitcore)
3 - Project specific clases

E.g:

```
var util = require('util');
var Promise = require('bluebird');
var EventEmitter = require('events').EventEmitter;

var bitcore = require('bitcore');
var Networks = bitcore.Networks;
var $ = bitcore.util.preconditions;
var _ = bitcore.deps._;
var p2p = require('bitcore-p2p');
var Peer = p2p.Peer;
var Messages = p2p.Messages;

var EventBus = require('./eventbus');
```

File: test/networkmonitor.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    };
+  });
+
+  it('instantiates correctly from constructor', function() {
+    var nm = new NetworkMonitor(busMock, peerMock);
+    should.exist(nm);
+  });
+
+  it('instantiates correctly from create', function() {
+    var nm = NetworkMonitor.create(busMock);
```
Comment (yemel) [inline test/networkmonitor.js]: Could you test that `opts` are being used to create a Peer object

Comment (maraoz) [inline test/networkmonitor.js]: Done: https://github.com/maraoz/bitcore-node/commit/df81629fa1373a40780e73cde70e4d0faffcdd45#diff-1e4575b10b51e27e8d3567a1c7edc176R53

**Comment by eordano** [comment]
LGTM

@yemel merge if no further comments

**Comment by yemel** [comment]
ACK

**Pull request merged**